Links open in Edge instead of my default browser

### Apache NetBeans version

Apache NetBeans 19

### What happened

URLs open in Edge, not in my default browser (Vivaldi)


### How to reproduce

All URLs seem to be affected. The simplest way to reproduce is Help->Online Support

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 10 

### JDK

openjdk 19.0.2 2023-01-17 OpenJDK Runtime Environment (build 19.0.2+7-44) OpenJDK 64-Bit Server VM (build 19.0.2+7-44, mixed mode, sharing)

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

I realize this may be very difficult due to Microsoft's continuing attempts to ignore my wishes on browser usage.  I will, however, note that Libre Office Writer opens its online help in my default browser.

### Are you willing to submit a pull request?

No

That didn't work:

`C:\Program Files\NetBeans-19\netbeans\bin>netbeans -Jorg.netbeans.modules.extbrowser.UseDesktopBrowse=true

The launcher has determined that the parent process has a console and will reuse it for its own console output.
Closing the console will result in termination of the running program.
Use '--console suppress' to suppress console output.
Use '--console new' to create a separate console window.
WARNING: package com.apple.eio not in java.desktop
WARNING: package sun.awt.X11 not in java.desktop
WARNING: package com.sun.java.swing.plaf.gtk not in java.desktop
Error: Could not find or load main class org.netbeans.modules.extbrowser.UseDesktopBrowse=true
Caused by: java.lang.ClassNotFoundException: org.netbeans.modules.extbrowser.UseDesktopBrowse=true`

--- a/ide/extbrowser/src/org/netbeans/modules/extbrowser/ExtBrowserImpl.java
+++ b/ide/extbrowser/src/org/netbeans/modules/extbrowser/ExtBrowserImpl.java
@@ -35,10 +35,7 @@ public abstract class ExtBrowserImpl extends HtmlBrowser.Impl {
 
     private static final RequestProcessor RP = new RequestProcessor(ExtBrowserImpl.class);
 
-    /** Lookup of this {@code HtmlBrowser.Impl}.  */
-    private Lookup lookup;
-
-    /** standart helper variable */
+    /** standard helper variable */
     protected PropertyChangeSupport pcs;
 
     /** requested URL */
@@ -82,9 +79,7 @@ public abstract class ExtBrowserImpl extends HtmlBrowser.Impl {
     @Override
     public void stopLoading() { }
     
-    protected void setTitle (String title) {
-        return;
-    }
+    protected void setTitle (String title) { }
     
     @Override
     public String getTitle() {
@@ -135,11 +130,8 @@ public abstract class ExtBrowserImpl extends HtmlBrowser.Impl {
     }
 
     protected final void loadURLInBrowser(final URL url) {
-        RP.post(new Runnable() {
-            @Override
-            public void run() {
-                loadURLInBrowserInternal(url);
-            }
+        RP.post(() -> {
+            loadURLInBrowserInternal(url);
         });
     }
 
@@ -179,6 +171,7 @@ public abstract class ExtBrowserImpl extends HtmlBrowser.Impl {
         pcs.removePropertyChangeListener (l);
     }
 
+    @Override
     public final Lookup getLookup() {
         return Lookup.EMPTY;
     }

--- a/ide/extbrowser/src/org/netbeans/modules/extbrowser/NbDdeBrowserImpl.java
+++ b/ide/extbrowser/src/org/netbeans/modules/extbrowser/NbDdeBrowserImpl.java
@@ -37,6 +37,7 @@ import org.openide.util.NbBundle;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.nio.file.Files;
 import java.util.logging.Level;
 import org.openide.util.Exceptions;
 
@@ -413,7 +414,7 @@ public class NbDdeBrowserImpl extends ExtBrowserImpl {
             while ((f == null) && (retries > 0)) {
                 retries--;
                 try {
-                    f = File.createTempFile("extbrowser", ".html");             // NOI18N
+                    f = Files.createTempFile("extbrowser", ".html").toFile();             // NOI18N
                     logFine("file:", f); // NOI18N
                     if (f != null) { 
                         fw = new FileWriter(f);

--- a/ide/extbrowser/src/org/netbeans/modules/extbrowser/SystemDefaultBrowser.java
+++ b/ide/extbrowser/src/org/netbeans/modules/extbrowser/SystemDefaultBrowser.java
@@ -28,7 +28,6 @@ import java.net.URL;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-import org.netbeans.modules.extbrowser.PrivateBrowserFamilyId;
 import org.openide.awt.HtmlBrowser;
 import org.openide.execution.NbProcessDescriptor;
 import org.openide.util.NbBundle;
@@ -44,32 +43,18 @@ public class SystemDefaultBrowser extends ExtWebBrowser {
 
     private static final long serialVersionUID = -7317179197254112564L;
     private static final Logger logger = Logger.getLogger(SystemDefaultBrowser.class.getName());
-    private static RequestProcessor RP = new RequestProcessor(SystemDefaultBrowser.class.getName(), 3);
+    private static final RequestProcessor RP = new RequestProcessor(SystemDefaultBrowser.class.getName(), 3);
     private transient AtomicBoolean detected = new AtomicBoolean(false);
 
-    private static final boolean ACTIVE;
-    static {
-        if (Boolean.getBoolean("org.netbeans.modules.extbrowser.UseDesktopBrowse")) {
-            if (Boolean.getBoolean("java.net.useSystemProxies") && Utilities.isUnix()) {
-                // remove this check if JDK's bug 6496491 is fixed or if we can assume ORBit >= 2.14.2 and gnome-vfs >= 2.16.1
-                logger.log(Level.FINE, "Ignoring java.awt.Desktop.browse support to avoid hang from #89540");
-                ACTIVE = false;
-            } else {
-                ACTIVE = Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE);
-            }
-        } else {
-            ACTIVE = false;
-        }
-    }
+    private static final boolean USE_JDK_BROWSER = Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)
+            && Boolean.getBoolean("org.netbeans.modules.extbrowser.UseDesktopBrowse");
 
     /** Determines whether the browser should be visible or not
      *  @return true when OS is not Windows and is not Unix with Default Browser capability and Desktop is inactive.
      *          false in all other cases.
      */
     public static Boolean isHidden() {
-        return Boolean.valueOf(
-                (!Utilities.isWindows() && !defaultBrowserUnixReady() && !Utilities.isMac())
-                && !ACTIVE);
+        return !Utilities.isWindows() && !defaultBrowserUnixReady() && !Utilities.isMac() && !USE_JDK_BROWSER;
     }
 
     private static boolean defaultBrowserUnixReady() {
@@ -88,13 +73,13 @@ public class SystemDefaultBrowser extends ExtWebBrowser {
      */
     @Override
     public HtmlBrowser.Impl createHtmlBrowserImpl() {
-        if (ACTIVE) {
-            return new Jdk6BrowserImpl();
+        if (USE_JDK_BROWSER) {
+            return new JdkBrowserImpl();
         } else if (Utilities.isWindows()) {
             return new NbDdeBrowserImpl(this);
         } else if (Utilities.isMac()) {
             return new MacBrowserImpl(this);
-        } else if (Utilities.isUnix() && !Utilities.isMac()) {
+        } else if (Utilities.isUnix()) {
             return new NbDefaultUnixBrowserImpl(this);
         } else {
             throw new UnsupportedOperationException(NbBundle.getMessage(SystemDefaultBrowser.class, "MSG_CannotUseBrowser"));
@@ -121,7 +106,7 @@ public class SystemDefaultBrowser extends ExtWebBrowser {
     }
 
     /** Default command for browser execution.
-     * Can be overriden to return browser that suits to platform and settings.
+     * Can be overridden to return browser that suits to platform and settings.
      *
      * @return process descriptor that allows to start browser.
      */
@@ -132,7 +117,7 @@ public class SystemDefaultBrowser extends ExtWebBrowser {
                 "{" + ExtWebBrowser.UnixBrowserFormat.TAG_URL + "}", // NOI18N
                 ExtWebBrowser.UnixBrowserFormat.getHint());
         }
-        if (!Utilities.isWindows() || ACTIVE) {
+        if (!Utilities.isWindows() || USE_JDK_BROWSER) {
             return new NbProcessDescriptor("", ""); // NOI18N
         }
 
@@ -181,19 +166,16 @@ public class SystemDefaultBrowser extends ExtWebBrowser {
         final HtmlBrowser.Impl impl = createHtmlBrowserImpl();
         final ExtBrowserImpl extImpl = impl instanceof ExtBrowserImpl ? (ExtBrowserImpl)impl : null;
         if (extImpl != null) {
-            RP.post(new Runnable() {
-                @Override
-                public void run() {
-                    setPrivateBrowserFamilyId(extImpl.detectPrivateBrowserFamilyId());
-                }
+            RP.post(() -> {
+                setPrivateBrowserFamilyId(extImpl.detectPrivateBrowserFamilyId());
             });
         }
     }
 
-    private static final class Jdk6BrowserImpl extends ExtBrowserImpl {
+    private static final class JdkBrowserImpl extends ExtBrowserImpl {
 
-        public Jdk6BrowserImpl() {
-            assert ACTIVE;
+        public JdkBrowserImpl() {
+            assert USE_JDK_BROWSER;
         }
 
         @Override

